### ISSUE TITLE
Helix machines should reboot post-workitem when exit code is STATUS_DLL_INIT_FAILED (-1073741502)

### ISSUE THREAD

**Issue by MattGal** [dotnet/arcade#10013]
As noted in https://github.com/dotnet/runtime/pull/72142, sometimes Windows ARM64 machines get into a state where every process started by the Helix client gets STATUS_DLL_INIT_FAILED (-1073741502)

To make things better for our users we should 
- [x] Introduce a final "exit code" message by default in the work item
- [ ] If the machine is a windows machine and the exit code is -1073741502, we should use existing functionality to cause a machine reboot post test- this seems to unblock the problem immediately.

**Comment by MattGal** [comment]
I'll do this work next week when I'm on FR, assigning to self for dibs.

**Comment by MattGal** [comment]
Merged https://dnceng.visualstudio.com/internal/_git/dotnet-helix-machines/pullrequest/24367, will do some testing when this finishes deploying to staging.

**Comment by MattGal** [comment]
Changes went in but I missed something; the exit code is treated as unsigned int inside the helix client (and magically becomes signed somewhere downstream)  so I've changed:

- Print code that puts exit code at end of log now converts to signed (to prevent confusion)
- Use unsigned value (added comment) when checking for infra retry / reboot behavior

New PR: https://dnceng.visualstudio.com/internal/_git/dotnet-helix-machines/pullrequest/24389?_a=files

**Comment by MattGal** [comment]
Latest changes solve the problem and make the logs look good, moving to rollout.

**Comment by MattGal** [comment]
These changes are now live and haven't caused trouble, so closing this issue.
